unit_test/test_spdm_callback/spdm_cert_verify_callback.c

```diff
@@ -0,0 +1,284 @@
+    /*type*/
+    result = libspdm_asn1_get_tag(&ptr, end, &obj_len,
+                                  LIBSPDM_CRYPTO_ASN1_CONTEXT_SPECIFIC + 9);
+    if (result) {
+        if ((obj_len != sizeof(m_libspdm_dice_tcbinfo_type)) ||
+            (!libspdm_consttime_is_mem_equal(ptr, m_libspdm_dice_tcbinfo_type, obj_len))) {
+            return false;
+        }
+        ptr += obj_len;
+    } else {
+        return false;
+    }
+    /*flagMask*/
+    result = libspdm_asn1_get_tag(&ptr, end, &obj_len,
+                                  LIBSPDM_CRYPTO_ASN1_CONTEXT_SPECIFIC + 10);
+    if (result) {
+        ptr += obj_len;
+    }
+
+    if (ptr == end) {
+        return true;
+    } else {
+        return false;
+    }
+}
+
+/*callback function for verifying cert_chain DiceTcbInfo extension*/
+bool libspdm_verify_spdm_cert_chain_with_dice(void *spdm_context, uint8_t slot_id,
+                                              size_t cert_chain_size, const void *cert_chain,
+                                              const void **trust_anchor,
+                                              size_t *trust_anchor_size)
+{
+    bool result;
+    libspdm_context_t *context;
+    const uint8_t *cert_chain_data;
+    size_t cert_chain_data_size;
+    size_t hash_size;
+    uint8_t *ptr;
+    uint8_t *tem_ptr;
+    int32_t length;
+    size_t obj_len;
+    uint8_t *end;
+    size_t cert_dice_tcb_info_size;
+    bool cert_chain_have_matched_dice;
+
+    /*verify peer cert chain integrity*/
+    result = libspdm_verify_peer_cert_chain_buffer_integrity(spdm_context, cert_chain,
+                                                             cert_chain_size);
+    if (!result) {
+        return false;
+    }
+
+    /*verify peer cert chain authority*/
+    result = libspdm_verify_peer_cert_chain_buffer_authority(spdm_context, cert_chain,
+                                                             cert_chain_size, trust_anchor,
+                                                             trust_anchor_size);
+    if (!result) {
+        return false;
+    }
+
+    context = spdm_context;
+    hash_size = libspdm_get_hash_size(context->connection_info.algorithm.base_hash_algo);
+
+    cert_chain_data = (const uint8_t *)cert_chain + sizeof(spdm_cert_chain_t) + hash_size;
+    cert_chain_data_size = cert_chain_size - sizeof(spdm_cert_chain_t) - hash_size;
+
+    length = (int32_t)cert_chain_data_size;
+    ptr = (uint8_t*)(size_t)cert_chain_data;
+    obj_len = 0;
+    end = ptr + length;
+    cert_chain_have_matched_dice = false;
+
+    while (ptr < end) {
+        cert_dice_tcb_info_size = 0;
+        tem_ptr = ptr;
+        result = libspdm_asn1_get_tag(&ptr, end, &obj_len,
+                                      LIBSPDM_CRYPTO_ASN1_SEQUENCE |
+                                      LIBSPDM_CRYPTO_ASN1_CONSTRUCTED);
+        if (result) {
+            /*verify Dice TCB info*/
+            result = libspdm_verify_cert_dicetcbinfo(tem_ptr, obj_len + (ptr - tem_ptr),
```

Comment:
If libspdm_verify_cert_dicetcbinfo returns FALSE and cert_dice_tcb_info_size!=0, tracing this function further, I think there is an error ("store buffer too small"- not sure what it meant).  Is it necessary to handle the error here?

Reply:
Thanks. 
Now `libspdm_verify_cert_dicetcbinfo` hard code the DiceTcbInfo buffer size, and the code categorizes the buffer error to invalid error.
https://github.com/DMTF/libspdm/blob/c1c1faa7fb59ac1879ed0f212ae83c48ad54baf5/unit_test/test_spdm_callback/spdm_cert_verify_callback.c#L56-L66

I don't think it is necessary to handle this issue.